cmd/service-controller/ports.go

```diff
@@ -0,0 +1,224 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+)
+
+type PortRange struct {
+	Start int
+	End   int
+}
+
+type FreePorts struct {
+	Available []PortRange
+}
+
+const (
```

Comment:
Yes, very good point! I have marked those in use for now - ideally it would take the values actually in use. That will be a little easier when the bridges are actually in the router config. This is certainly an area to improve on but I think it should suffice for now.